src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers/HttpContextCurrentAnalyzer.cs

```diff
@@ -40,12 +44,50 @@ public override void Initialize(AnalysisContext context)
             context.EnableConcurrentExecution();
             context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
 
-            context.RegisterSyntaxNodeAction(AnalyzeMemberAccessExpressions, SyntaxKind.SimpleMemberAccessExpression);
+            context.RegisterCompilationStartAction(context =>
+            {
+                if (context.Compilation.Language == LanguageNames.CSharp)
+                {
+                    context.RegisterSyntaxNodeAction(AnalyzeMemberAccessExpressionsCsharp, CS.SyntaxKind.SimpleMemberAccessExpression);
+                }
+                else if (context.Compilation.Language == LanguageNames.VisualBasic)
+                {
+                    context.RegisterSyntaxNodeAction(AnalyzeMemberAccessExpressionsVb, VB.SyntaxKind.SimpleMemberAccessExpression);
+                }
+            });
+        }
+
+        private void AnalyzeMemberAccessExpressionsCsharp(SyntaxNodeAnalysisContext context)
+        {
+            var memberAccessExpression = (CSSyntax.MemberAccessExpressionSyntax)context.Node;
+
+            // If the accessed member isn't named "Current" bail out
+            if (!TargetMember.Equals(memberAccessExpression.Name.ToString(), StringComparison.Ordinal))
+            {
+                return;
+            }
+
+            // If the call is to a method called Current then bail out since they're
+            // not using the HttpContext.Current property
+            if (memberAccessExpression.Parent is CSSyntax.InvocationExpressionSyntax)
```

Comment:
nit: try to put string comparisons later in the analyzer "bail out" flow. I don't have a doc recommending this but I'm basing that on string comparison always being slower than other conditionals we can run.

Reply:
The best comparison we have here is if Current is not the name, then we shouldn't continue at all. It's the best filter I can think of